baseplate/clients/thrift.py

```diff
@@ -166,6 +169,21 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                     if span.flags:
                         prot.trans.set_header(b"Flags", str(span.flags).encode())
 
+                    min_timeout = None
+                    is_time_retry_policy = isinstance(retry_policy, TimeBudgetRetryPolicy) or (
```

Comment:
The retry policies are an internal implementation detail here. It's a bit weird to expose them directly.  If there's a time_remaining value, would we ever _not_ want to propagate that as a deadline? It sounds like y'all ran into some other number coming out of that iterator which is surprising to me. What kind of data did you see there instead of a time or None?

Reply:
Oh you are right. I read the implementation of `MaximumAttemptsRetryPolicy` wrong. I thought it emits number of retries.

So yes it's either None or time remaining, and those are not bugs 😅

Reply:
heh, it took me too a lot of staring at that code to realize it was still returning time. i think there's an opportunity for readability improvement there.

Reply:
https://github.com/reddit/baseplate.py/pull/577